src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,60 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie session " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a movie session by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> getAll(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getSessions = session.createQuery("from MovieSession ms where "
+                    + "ms.movie.id =:movieId AND ms.showTime = :showTime ", MovieSession.class);
+            getSessions.setParameter("movieId", movieId);
+            getSessions.setParameter("showTime", date);
+            return getSessions.getResultList();
```

Comment:
The current implementation of `getAll` method in `MovieSessionDaoImpl` is not following the best practices for querying data. The method should not get all `MovieSession` from dao and then sort them on service layer in `findAvailableSessions()`. It's better for performance to extract "ready to use" data from DB on dao layer. Additionally, the method name `getAll` is misleading as it suggests retrieving all movie sessions, whereas it should be filtering by movieId and a specific date. Consider renaming the method to better reflect its purpose and ensure the query fetches sessions for the given date within the day's duration, not just by the exact timestamp.